### ISSUE TITLE
OAuth Not OIDC

### ISSUE THREAD

**Issue by billcunnien** [DuendeSoftware/Support#270]
**Which version of Duende IdentityServer are you using?**
6

**Which version of .NET are you using?**
.NET Core 6

**Describe the bug**

The workflow that I want to implement is for an interactive application that demands a two-step authorization flow - /authorize (with redirect to a login page) and /token (to exchange that code received after login for the access token used with the API). I do not want OIDC which seems to combine things into one call. Personally, I think that is the better option, but the vendor does not support it. Your examples are not clear about how to make this happen. Can you point me to a clear example? Or, can you help me resolve the exception - The oauth state was missing or invalid?

**To Reproduce**

My shot in the dark regarding the Web Client program.cs setup:

builder.Services.AddAuthentication(options =>
    {
// TODO: no idea what to put here!
        options.DefaultChallengeScheme = "oauth";
    })
    .AddOAuth("oauth", options =>
    {
        options.AuthorizationEndpoint = "https://localhost:5001/connect/authorize"; // TODO: no parameters?
        options.CallbackPath = new PathString("/"); // TODO: not clear what really should be here!

        options.ClientId = "vendor";
        options.ClientSecret = "secret";

        options.Scope.Add("openid");
        options.Scope.Add("profile");
        options.Scope.Add("API");

        options.TokenEndpoint = "https://localhost:5001/connect/token";
    });

**Expected behavior**

User gets redirected to the login page where they proceed to login, then they should get sent back to the web app so that it can retrieve the token using a backend call.

**Log output/exception with stacktrace**

N/A

**Additional context**

Just to be clear, I have the OIDC stuff working just fine - I get a working access token in the payload. I need to retool this to be OAuth only using the two-step flow (authorization code/token exchange).

Thanks, as always, for your help!!
Bill

**Comment by billcunnien** [comment]
Thanks!

Using the web client project from the quickstarts, I adjusted the AddAuthentication method to include those two settings:

options.ResponseType = "code";
options.ResponseMode = "query";

I adjusted the client ID and secret, too. My IS is running and this web client fires up properly, redirects to the login page where I can login successfully, then gets sent back to the web application displaying a claims/properties page. There is no code in the query string. The access token is clearly displayed in the HTML. So, that seems to work just fine. I would have fully expected a code, though, not the token. Also, I do not see any evidence of a refresh token. I will go ahead with this setup, but I am not at all confident that it will work for our vendor.
